base/bucket.go

```diff
@@ -112,6 +120,49 @@ func (spec BucketSpec) IsWalrusBucket() bool {
 	return strings.Contains(spec.Server, "walrus:")
 }
 
+func (spec BucketSpec) IsTLS() bool {
+	return strings.HasPrefix(spec.Server, "couchbases") || strings.HasPrefix(spec.Server, "https")
+}
+
+func (spec BucketSpec) UseClientCert() bool {
+	if spec.Certpath == "" || spec.Keypath == "" {
+		return false
+	}
+	return true
+}
+
+// Builds a gocb connection string based on BucketSpec.Server.
+// Adds idle connection configuration, and X.509 auth settings when
+// certpath/keypath/cacertpath specified.
+func (spec BucketSpec) GetGoCBConnString() (string, error) {
+
+	connSpec, err := gocbconnstr.Parse(spec.Server)
+	if err != nil {
+		return "", err
+	}
+
+	// Increase the number of idle connections per-host to fix SG #3534
+	if connSpec.Options == nil {
+		connSpec.Options = map[string][]string{}
+	}
+	asValues := url.Values(connSpec.Options)
```

Comment:
Is this PR is currently based on https://github.com/couchbase/sync_gateway/pull/3545 rather than master?  Looks like it might need a rebase off of master.

Reply:
Have rebased on master.

Reply:
Ok I see why I thought this wasn't based on master .. this code is the same as the code from 

https://github.com/couchbase/sync_gateway/blob/a449b3b569049d8187fd26ea99e1f622c2948220/base/bucket_gocb.go#L118-L121

I'd say it would be better to to centralize this logic

Reply:
Right - that was my thinking behind moving that handling to BucketSpec - so all the conn string work was in the same location.